example/scalalib/module/15-contrib-checkstyle/build.mill

```diff
@@ -0,0 +1,22 @@
+package build
+import mill._, scalalib._
+import $ivy.`com.lihaoyi::mill-contrib-checkstyle:`
+
+import mill.contrib.checkstyle._
+import com.etsy.sbt.checkstyle.CheckstyleSeverityLevel
+
+object `package` extends RootModule with CheckstyleModule {
+  def checkstyleVersion = "9.3"
+  def checkstyleSeverityLevel = Some(CheckstyleSeverityLevel.Info)
+}
+
+/** Usage
+
+> ./mill checkstyle # Run checkstyle to produce a report, optionally failing the build if violations are severe enough
+Checkstyle warning found in ...src/InputFileName1.java:2: Top-level class MyAnnotation1 has to reside in its own source file.
+Checkstyle warning found in ...src/InputFileName1.java:13: Top-level class Enum1 has to reside in its own source file.
+Checkstyle warning found in ...src/InputFileName1.java:26: Top-level class TestRequireThisEnum has to reside in its own source file.
+3 issue(s) found in Checkstyle report: ...out/checkstyle.dest/checkstyle-report.xml
+error: ...Checkstyle issues found
```

Comment:
Let's flesh this out a bit more:

- Assert on a handful of different checkstyle errors, not just `Top-level class has to reside in its own source file`
- Let's exercise the process of fixing the errors: we can run `sed` to mangle the files (see other example tests), and run `checkstyle` again and verify that it no longer errors

Reply:
I agree with fixing and checking it passes, have extended the test to cover that part.

What sort of different errors do we want to cover? I don't see a lot of value in duplicating the whole checkstyle checksuite, we're not really exercising the plugin code more by having more than a couple of errors (as long as we can pick up more than 1 error reported by checkstyle, I can't see any reasonable way for the plugin to break on a different error), but happy to add any you think are important.